go-controller/pkg/retry/obj_retry.go

```diff
@@ -710,6 +717,9 @@ func (r *RetryFramework) WatchResourceFiltered(namespaceForFilteredHandler strin
 				})
 			},
 			DeleteFunc: func(obj interface{}) {
+				if r.ResourceHandler.FilterOutResource(obj) {
```

Comment:
Just giving visibility that this can be an object of type `DeletedFinalStateUnknown`. I guess we are not handling this correctly in general with the retry framework so not asking anything for this PR, but if we are not sure of the state of the object when it was deleted, we should try to process it anyway?

Reply:
Good point, FilterOutResource should not filter those out and the controller should get a chance to handle it.